### PR TITLE
cleanup-server

### PR DESCRIPTION
**Pull request by Jurgenvs**
Cleaned up the server and client and also cleaned up the Local storage.

### CODE REVIEW

**Comment by sirpolomotro** [comment]
Looks good, however client still has too much responsibilities. It would be better (like you said your self) to have something like a QueryCreater

File: core/src/cg/group4/game_logic/StandUp.java [hunk 1]
```diff
@@ -31,11 +32,6 @@
     protected static StandUp cInstance;
 
     /**
-     * Background music to be played.
-     */
-    protected Music cBackgroundMusic;
```
Comment (sirpolomotro) [inline core/src/cg/group4/game_logic/StandUp.java]: Good! Now also the sensor reader and acceleration status.
Wouldn't it be better to have a class that keeps track of such object, a so called container class?

File: core/src/cg/group4/util/audio/AudioPlayer.java [hunk 2]
```diff
@@ -94,7 +95,7 @@ protected final void afterChange() {
         if(!cAudioEnabled){
             cLastPlayed.stop();
         } else {
-            AudioPlayer.getInstance().playAudio(StandUp.getInstance().getBackGroundMusic());
+//            AudioPlayer.getInstance().playAudio(StandUp.getInstance().getBackGroundMusic());
```
Comment (sirpolomotro) [inline core/src/cg/group4/util/audio/AudioPlayer.java]: Why is this line commented away..?
Should be fixed.

File: core/src/cg/group4/view/screen_mechanics/ScreenStore.java [hunk 3]
```diff
@@ -60,10 +60,10 @@ public static ScreenStore getInstance() {
      */
     public void init() {
         addScreen("Home", new HomeScreen());
-        addScreen("Collection", new CollectiblesScreen());
```
Comment (sirpolomotro) [inline core/src/cg/group4/view/screen_mechanics/ScreenStore.java]: Collectible screen should be fixed for local/remote.

File: core/src/cg/group4/view/screen_mechanics/ScreenStore.java [hunk 4]
```diff
@@ -60,10 +60,10 @@ public static ScreenStore getInstance() {
...
+//        addScreen("Collection", new CollectiblesScreen());
         addScreen("Settings", new SettingsScreen());
         addScreen("Network", new NetworkScreen());
-        addScreen("Groups", new GroupScreen());
+//        addScreen("Groups", new GroupScreen());
```
Comment (sirpolomotro) [inline core/src/cg/group4/view/screen_mechanics/ScreenStore.java]: Groups should only be made/visisble if a remote server is available

File: desktop/src/cg/group4/server/DesktopIDResolver.java [hunk 5]
```diff
@@ -1,4 +1,6 @@
-package cg.group4.client;
+package cg.group4.server;
+
```
Comment (sirpolomotro) [inline desktop/src/cg/group4/server/DesktopIDResolver.java]: Why has this been moved to the server..?

File: server/src/cg/group4/server/database/ResponseHandler.java [hunk 6]
```diff
@@ -1,5 +1,13 @@
 package cg.group4.server.database;
```
Comment (sirpolomotro) [inline server/src/cg/group4/server/database/ResponseHandler.java]: Nice solution!

File: server/src/cg/group4/server/database/query/CreateGroup.java [hunk 7]
```diff
@@ -3,57 +3,78 @@
...
+                        resultSet.getString("Name"),
+                        resultSet.getString("OwnerId"),
+                        resultSet.getString("Username")
+                );
+            }
 
-        Group group = new Group(Integer.toString(groupData.getGroupId()), groupData);
-        PlayerData playerData = new PlayerData(cOwnerId);
-        playerData.setUsername(null);
+            group = new Group(Integer.toString(groupData.getGroupId()), groupData);
```
Comment (sirpolomotro) [inline server/src/cg/group4/server/database/query/CreateGroup.java]: `groupData.getGroupId()` could be replaced with `groupId`.

File: server/src/cg/group4/server/database/query/CreateGroup.java [hunk 8]
```diff
@@ -3,57 +3,78 @@
...
-        );
+            try (ResultSet resultSet = statement.executeQuery()) {
+                groupId = resultSet.getInt("GroupId");
 
-        resultSet.close();
-        statement.close();
+                groupData = new GroupData(
+                        groupId,
+                        resultSet.getString("Name"),
+                        resultSet.getString("OwnerId"),
```
Comment (sirpolomotro) [inline server/src/cg/group4/server/database/query/CreateGroup.java]: `resultSet.getString("Name"), resultSet.getString("OwnerId"),` could be replace with `cGroupName, cOwnerId`

File: server/src/cg/group4/server/database/query/GetGroup.java [hunk 9]
```diff
@@ -22,28 +22,30 @@
...
      */
-    public GetGroup(String groupId) {
+    public GetGroup(final String groupId) {
         cGroupId = groupId;
     }
 
     @Override
-    public Group query(DatabaseConnection databaseConnection) throws SQLException {
+    public Group query(final Connection databaseConnection) throws SQLException {
+        String preparedStatement = "SELECT * FROM 'Group' INNER JOIN User ON 'Group'.OwnerId = User.Id "
```
Comment (sirpolomotro) [inline server/src/cg/group4/server/database/query/GetGroup.java]: Owner data is now stored on the local db. So this wont work... remote shouldn't even have a user table.. Instead groups should contain both the owners id and name.

File: server/src/cg/group4/server/database/query/GetGroupData.java [hunk 10]
```diff
@@ -14,22 +14,24 @@
...
-        while (resultSet.next()) {
-            list.add(new GroupData(
-                    resultSet.getInt("GroupId"),
-                    resultSet.getString("Name"),
-                    resultSet.getString("OwnerId"),
-                    resultSet.getString("Username")
-            ));
+
+        String query = "SELECT G.Key AS GroupId, G.Name AS Name, G.OwnerId AS OwnerId, U.Username AS Username, U.Id "
+                + "FROM 'Group' G INNER JOIN User U ON G.OwnerId = U.Id";
```
Comment (sirpolomotro) [inline server/src/cg/group4/server/database/query/GetGroupData.java]: Same here... Remote server should not contain user data... and local database should not contain group data...

File: server/src/cg/group4/server/database/query/RequestHostCode.java [hunk 11]
```diff
@@ -28,24 +30,33 @@ public RequestHostCode(final String ip) {
...
-        do {
-            code = rnd.nextInt(cMaxCode);
-            ResultSet resultSet = statement.executeQuery("SELECT * FROM Event_Hosts WHERE Code = " + code);
-            exists = resultSet.isBeforeFirst();
-            resultSet.close();
+        String preparedQuery = "SELECT * FROM Event_Hosts WHERE Code = ?";
+
+        try (PreparedStatement statement = databaseConnection.prepareStatement(preparedQuery)) {
+            do {
+                code = rnd.nextInt(MAXCODE);
```
Comment (sirpolomotro) [inline server/src/cg/group4/server/database/query/RequestHostCode.java]: Couldn't this be moved to the deceleration of `code`.

File: server/src/cg/group4/server/database/query/RequestPlayerData.java [hunk 12]
```diff
@@ -25,35 +24,45 @@
      * @param id The id of the player to be retrieved.
      */
     public RequestPlayerData(final String id) {
-        cId = id;
```
Comment (sirpolomotro) [inline server/src/cg/group4/server/database/query/RequestPlayerData.java]: Good improvement

File: server/src/cg/group4/server/database/query/ResetPlayerData.java [hunk 13]
```diff
@@ -12,24 +11,25 @@
...
-    public Serializable query(DatabaseConnection databaseConnection) throws SQLException {
-        Statement statement = databaseConnection.query();
-        statement.executeUpdate("DELETE FROM Collectible WHERE GroupId = '" + cId + "'");
-        databaseConnection.commit();
-
-        statement.executeUpdate("DELETE FROM User WHERE Id = '" + cId + "'");
-        databaseConnection.commit();
-
-        statement.close();
+    public Serializable query(final Connection databaseConnection) throws SQLException {
```
Comment (sirpolomotro) [inline server/src/cg/group4/server/database/query/ResetPlayerData.java]: You no longer remove the player data out of the user table.. Either rename the query or reset the player data in the user table (could also only be the timers)

File: server/src/cg/group4/server/database/query/UpdatePlayerData.java [hunk 14]
```diff
@@ -25,39 +25,61 @@
...
+    }
 
-        return null;
+    /**
+     * Updates the players data.
+     * @param column The column to update.
+     * @param newValue The new value.
+     * @throws SQLException Throws if something went wrong while updating.
+     */
+    protected void updateData(final String column, final long newValue) throws SQLException {
```
Comment (sirpolomotro) [inline server/src/cg/group4/server/database/query/UpdatePlayerData.java]: Could this not be incorperated into the `updateData` above.

File: server/src/cg/group4/util/IpResolver.java [hunk 15]
```diff
@@ -14,15 +15,18 @@
     /**
      * String to be used for the URL to check external IP.
      */
-    protected final String cVerifyURl = "http://checkip.amazonaws.com";
+    private static String cVerify = "http://checkip.amazonaws.com";
```
Comment (sirpolomotro) [inline server/src/cg/group4/util/IpResolver.java]: Why did you change this to private..?

File: core/src/cg/group4/game_logic/StandUp.java (hunk 1 above)
Comment (Jurgenvs) [inline core/src/cg/group4/game_logic/StandUp.java]: Not something for this branch, but has to be done. I'd suggest creating an Issue and assigning it to the person that created it.

File: core/src/cg/group4/view/screen_mechanics/ScreenStore.java (hunk 3 above)
Comment (Jurgenvs) [inline core/src/cg/group4/view/screen_mechanics/ScreenStore.java]: This is something that has to be fixed in general. Screens do not properly work with the remote server yet. They now have a hard dependency on the remote server, but what if the player has no WIFI? Screens have to be fixed in another branch.

File: core/src/cg/group4/view/screen_mechanics/ScreenStore.java (hunk 4 above)
Comment (Jurgenvs) [inline core/src/cg/group4/view/screen_mechanics/ScreenStore.java]: This is something that has to be fixed in general. Screens do not properly work with the remote server yet. They now have a hard dependency on the remote server, but what if the player has no WIFI? Screens have to be fixed in another branch.

File: desktop/src/cg/group4/server/DesktopIDResolver.java (hunk 5 above)
Comment (Jurgenvs) [inline desktop/src/cg/group4/server/DesktopIDResolver.java]: It was first in the Client module, but it does not belong there (since it is a DesktopIDResolver). I moved it to the proper module, but will make a package in there called client, since it is something involving the client. Also reflected this change in Android.

File: server/src/cg/group4/server/database/ResponseHandler.java (hunk 6 above)
Comment (Jurgenvs) [inline server/src/cg/group4/server/database/ResponseHandler.java]: It was to fix the Android SQLite server. SQLDroid wouldn't function with multiple connections. Decided to have 1 connection that has many statements.

File: server/src/cg/group4/server/database/query/CreateGroup.java (hunk 7 above)
Comment (Jurgenvs) [inline server/src/cg/group4/server/database/query/CreateGroup.java]: Fixed

File: server/src/cg/group4/server/database/query/CreateGroup.java (hunk 8 above)
Comment (Jurgenvs) [inline server/src/cg/group4/server/database/query/CreateGroup.java]: Fixed

File: server/src/cg/group4/server/database/query/GetGroup.java (hunk 9 above)
Comment (Jurgenvs) [inline server/src/cg/group4/server/database/query/GetGroup.java]: Queries should in general be looked on once again, now that we have Remote and Local storage. We'll need to discuss what we are going to store where. Not for this PR.

File: server/src/cg/group4/server/database/query/GetGroupData.java (hunk 10 above)
Comment (Jurgenvs) [inline server/src/cg/group4/server/database/query/GetGroupData.java]: Queries should in general be looked on once again, now that we have Remote and Local storage. We'll need to discuss what we are going to store where. Not for this PR.

File: server/src/cg/group4/server/database/query/RequestPlayerData.java (hunk 12 above)
Comment (Jurgenvs) [inline server/src/cg/group4/server/database/query/RequestPlayerData.java]: Did that to seperate the insertUser to a new method, but works better in general ;)

File: server/src/cg/group4/server/database/query/ResetPlayerData.java (hunk 13 above)
Comment (Jurgenvs) [inline server/src/cg/group4/server/database/query/ResetPlayerData.java]: My opinion was that the user could remain in the database, but only the things that are 'obtained' need to be removed. Otherwise we'd have to immediately resubmit our player data in the database. This is something we can look at once we go over all the queries and screens.

File: server/src/cg/group4/server/database/query/UpdatePlayerData.java (hunk 14 above)
Comment (Jurgenvs) [inline server/src/cg/group4/server/database/query/UpdatePlayerData.java]: Agree that it was hugely duplicate. Created these methods before later finding out that it's also possible to pass Objects to the PreparedStatement. Made it into a single method.

File: server/src/cg/group4/util/IpResolver.java (hunk 15 above)
Comment (Jurgenvs) [inline server/src/cg/group4/util/IpResolver.java]: Something with checkstyle or findbugs complaining. Changed it back to protected and did not see the change again, so no clue.

File: core/src/cg/group4/util/audio/AudioPlayer.java [hunk 16]
```diff
@@ -42,6 +47,11 @@ public AudioPlayer() {
         cAudioChangedSubject = new Subject();
         cPreferences = Gdx.app.getPreferences("AUDIO");
         cAudioEnabled = cPreferences.getBoolean("ENABLED", true);
+        cDefaultMusic = Gdx.audio.newMusic(Gdx.files.internal("music/Summer Day.mp3"));
+//        cDefaultMusic = Assets.getInstance().getMusic("music/Summer Day.mp3");
```
Comment (NickTUD) [inline core/src/cg/group4/util/audio/AudioPlayer.java]: Would be better to just remove commented code.

File: server/src/cg/group4/server/database/query/RequestCollection.java [hunk 17]
```diff
@@ -29,30 +29,30 @@ public RequestCollection(final String groupId) {
...
-                        resultSet.getString("OwnerId")));
-            } catch (ParseException e) {
-                e.printStackTrace();
+
+        String preparedQuery = "SELECT * FROM Collectible WHERE GroupId = ?";
+
+        try (PreparedStatement statement = databaseConnection.prepareStatement(preparedQuery)) {
+            statement.setString(1, cGroupId);
+            try (ResultSet resultSet = statement.executeQuery()) {
+                CollectibleFactory factory = new CollectibleFactory();
```
Comment (Surskit) [inline server/src/cg/group4/server/database/query/RequestCollection.java]: Is it neccesary create a new factory every time you query?

**Comment by Surskit** [comment]
Surskit approves!                                                               
                                                  ,                                 
                                                 ';                                 
                                                :`                                  
                                               .`'                                  
                                               ```                                  
                                              '`,                                   
                                             ``.;                                   
                                             ;..                                    
                                            ...,                                    
                                            ;..'                                    
                                            ...,                                    
                                           '...                                     
                                           ....                                     
                                          `...,                                     
                                      .',.....,`                                    
                                   .'`````.......;                                  
                                 .:.```````........:                                
                                #..`````;`..........;                               
                               ;.,..```.`.....:......:                              
                              ;``.'....,......'......::                             
                             ,.`....:'+.......'....;:..:                            
                             ,...##,....:;''''+':.......                            
                             ...#+ .....................:                           
                            ,...@@ ':...........`,#.....:                           
                           :+...#+@@...........#  @+.....                           
                         ;,,+..,.#@,...........@;@##.....`                          
                       .:,,,',,,,..............@@++,....,'                          
                      ',,,;`.,,,,,,.............+#:.....,:,;.                       
                     ',,,,`;,,,,,,,............,...,,..,,,,,,;.                     
                    ;,,'`;.,,,,,...............,,,,,,.,,:`::,,,'`                   
                   ',,;;...,:;;,,,..............,,,,.,,,,   `;,,.'                  
                  '.,+....'`   ;,,,,,.............,,,,,'      .:..:`                
                 ...'...;       .,,,,,,,,,,,,,,,,,,,,,,,'       '...+               
                 ,:...,.          ;,,,,,,,,,,,,,,,,,,',,,;       .,..'              
                ''...;              ;:,,,,,,,,,,,,',  ',,.`        +..,             
               `:..:                   ,'';''+;:`      '...         :..,            
              `...'                                     ,..;         ,..,           
             ,..,+                                       ...;         `..:          
            :..;.                                        :...`         `,,'         
           ;..+.'                                         ;..;           :,+        
          '..+..,                                          ...:           ,,;       
         '..:;.,                                           ,...            :,.      
        ;..` ','                                            :..'            ,,      
       ...`  :,,                                             ...`            ,;     
       ..:   ,,                                              '..'            `,+    
      :,'    ,,                                               ...`            .,:   
     ','     ,'                                               '..'             ,.   
    ,,:      ,'                                                ...              ''  
    ,,`      ,,                                                '..;              +: 
   ','       ,                                                  ...               : 
  .,;        ,                                                  ;..,               '
  ;,`        ,                                                  `..;                
 .,'         ,                                                   ;..                
 ;:          #                                                   ...+               
 ,:                                                               :.:               
;;                                                                ;,,               
:                                                                  ,,'              
                                                                   +,;              
                                                                   ,,,              
                                                                    ,,,             
                                                                    ',;             
                                                                    .,'             
                                                                     ,:             
                                                                     ',             
                                                                     .,             
                                                                      ,`            
                                                                      '`            
                                                                      .

**Comment by NickTUD** [comment]
Approved :+1:
